oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/StringMatchTest.java

```diff
@@ -23,44 +23,44 @@
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
-public class EqualMatchTest {
+public class StringMatchTest {
 
     @Test
     public void integerShouldEqualWhenLargerThan128() {
         Integer a = 334;
         Integer b = 334;
-        boolean match = new EqualMatch().match(a, b);
```

Comment:
From these tests, the original marcher should already support numeric types, just need to wrap primitives such as `int` to boxed type `Integrr` or add method signatures with primitive types

Reply:
Same reason as https://github.com/apache/skywalking/pull/7636#discussion_r701062504. And discussed at https://github.com/apache/skywalking/discussions/7517

Reply:
Am I trying below whether adding the method signature can be directly supported

Reply:
There is a question about what the regression tests need to do every time you modify the oap part of the code

Reply:
> Am I trying below whether adding the method signature can be directly supported

What is this? A question or something?

> There is a question about what the regression tests need to do every time you modify the oap part of the code

There is no guidance or standards about this, because this is rarely to change, and has been changed for a long time.

Reply:
For this problem kezhenxu94 raised a better solution I m trying

Reply:
![image](https://user-images.githubusercontent.com/45945752/131850965-c22bfa4f-2237-436e-b366-9b3313e98059.png)
As kezhenxu94 said this can be solved by simply the matching rules support numbers in stringmatch I think this is a better solution and is the class name restored to EqualMatch, I think EqualMatch is more reasonable in design and functionality

Reply:
whether it is working or not, is not important at all. If you want to more method, consider int and long. But you would not compare number, especially double and float. Because simply in Java `1.01 == 1.01` is not guaranteed.